lib/pbench/test/unit/server/test_requests.py

```diff
@@ -88,109 +88,139 @@ def test_json_object(client, caplog, server_config):
 
 
 class TestUpload:
+    @pytest.fixture
+    def setup_ctrl(self):
+        self.controller = socket.gethostname()
+        yield
+        self.controller = None
+
     @staticmethod
-    def test_missing_authorization_header(client, caplog, server_config):
-        response = client.put(
-            f"{server_config.rest_uri}/upload/ctrl/{socket.gethostname()}",
-            headers={"filename": "f.tar.xz"},
-        )
+    def gen_uri(server_config, filename="f.tar.xz"):
+        return f"{server_config.rest_uri}/upload/{filename}"
+
+    def gen_headers(self, auth_token, md5=None):
+        headers = {
+            "Authorization": "Bearer " + auth_token,
+            "controller": self.controller,
+        }
+        if md5:
+            headers["Content-MD5"] = md5
+        return headers
+
+    def test_missing_authorization_header(self, client, caplog, server_config):
+        response = client.put(self.gen_uri(server_config))
         assert response.status_code == 401
         for record in caplog.records:
-            assert record.levelname not in ("ERROR", "CRITICAL")
+            assert record.levelname not in ("DEBUG", "ERROR", "CRITICAL")
```

Comment:
Hmm. If we know that `DEBUG` level isn't enabled for the unit tests, then including it is pointless. Otherwise... I'm not sure we should ever disallow `DEBUG` to be used freely where a developer thinks it might be useful. This seems like more of a test of the `Logging` system (we shouldn't have debug enabled) than of the interface...

Reply:
I concur.  Moreover, I've been running the unit tests individually under my IDE, so I'm not aware of any reason why `DEBUG` level would be disabled in that case.  So, I second the notion that we should be permitting `DEBUG` to be used wherever we see fit and that the unit tests should not be applying back-pressure to that.

I'm not opposed to the unit tests checking the logs.  But they should only be checking for things which must or must not be there...and, generally speaking, I don't think that `"DEBUG"` should be in either list.

Reply:
If we make a code change that enabled `DEBUG` logging inadvertently, then we could potentially cause a server failure in production.

For agent side code I agree, but when it comes to the server, we want to know exactly what is being logged.

If we want to add `DEBUG` log entries to the code, the unit tests should then be run with `DEBUG` enabled so that we allow those.

But I have been on the operational end of production systems going down because somebody mistakenly emitted `DEBUG` logs in the worse possible place, but was never caught in testing.

The upload API is particularly critical because we can potentially have many more clients concurrently than operations performed from the browser.

Reply:
> I have been on the operational end of production systems going down because somebody mistakenly emitted `DEBUG` logs in the worse possible place, but was never caught in testing.

This is an excellent point.

However, is this assertion in the unit test actually going to catch that case?  E.g., if the logger used in unit test execution has `DEBUG` level disabled, then the test assertion won't detect the problem which would arise if `DEBUG` level logging were inadvertently enabled in production.

If we really want to avoid `DEBUG` level logging in production, then we probably need something like a Git hook which makes sure that it doesn't get merged into the source.  It's not clear to me that clogging up the unit tests with this gets us to where we need to be.